src/typechecker.erl

```diff
@@ -798,6 +798,8 @@ negate_num_type({type, P, union, Tys}) ->
     %% contain any unresolved types. So it is ok to normalize the result with an
     %% empty TEnv.
     normalize({type, P, union, [negate_num_type(Ty)||Ty <- Tys]}, #tenv{});
+negate_num_type(None = {type, _, none, []}) ->
+    None;
```

Comment:
It could be argued that we should report a type error here instead. What's your argument for returning none() instead of failing?

Reply:
No strong reasons. This was the minimal fix to make the crash go away. The call to `negate_num_type` is here: https://github.com/josefs/Gradualizer/blob/d01371f7fa110654fa3e93f08dbf2a61241daa36/src/typechecker.erl#L1462-L1470
To throw an error we'd need to special-case `Ty = none()`, and do the same for the cases above for `+`, `bnot`, and `not`.

Since this was just a minor bugfix PR, I didn't want to change the typing rules.

Reply:
This is interesting. If `none()` comes from some type calculation maybe it should cause a type error the first time it is encountered (or maybe at the end of the flow), but probably not in the middle (ie not in negate).

Otoh if `none()` comes straight from a type spec that reflects user intention, so that should not count as a type error.

It would be good to differentiate between the two cases, maybe in the user-spec case the user should call it `no_return()` (so it would have a special meaning, not just an alias for `none()`, similarly as `term()` vs `any()` distinction. I think dialyzer makes this distinction too). And `none()` would be always a type error.

Reply:
Right. For the sake of this PR I think it makes sense to leave the typing rules as is. 

@gomoripeti, your idea of differentiating `no_return()` and `none()` is interesting. But what if the programmer actually writes `none()` in a spec. What should the Gradualizer do then? I don't like the idea of rejecting the spec, there is legacy code which uses `none()`.

My sense is that it'd be good to get user feedback on this particular issue. It is not clear to me what the right type rule should be in the end. Leaving things as they are and erring on the side of being too lenient is the right thing to do right now.